packages/ui/react-ui-board/src/Grid/Grid.tsx

```diff
@@ -158,49 +173,65 @@ const RootInner = forwardRef<GridController, RootProps>(
             padding: margin,
           }}
         >
-          {outer}
-          {/* Board container. */}
-          <div
-            className={mx('relative transition-transform duration-300 border border-separator', zoom && 'scale-50')}
-            style={{
-              width: boardSize.width,
-              height: boardSize.height,
-            }}
-          >
-            {/* Scrollable container. */}
-            <div className={mx('absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2')}>
-              {inner}
-              {items.map((item, index) => (
-                <Tile
-                  item={item}
-                  key={index}
-                  layout={layout.tiles[item.id] ?? { x: 0, y: 0 }}
-                  onClick={() => controller.center(item.id)}
-                />
-              ))}
-            </div>
-          </div>
+          {children}
         </div>
       </GridContextProvider>
     );
   },
 );
 
-const Root = <T extends HasId = any>(props: RootProps<T>) => <RootInner {...props} />;
-
 Root.displayName = 'Grid.Root';
 
+//
+// Content
+//
+
+type ContentProps<T extends HasId = any> = ThemedClassName<
+  PropsWithChildren<{
+    items?: T[];
+  }>
+>;
+
+const Content = <T extends HasId = any>({ classNames, children, items }: ContentProps<T>) => {
+  const { layout, bounds, zoom } = useGridContext(Content.displayName);
+
+  return (
+    <div
+      className={mx(
+        'relative transition-transform duration-300 border border-separator rounded-lg',
+        zoom && 'scale-50',
+        classNames,
+      )}
+      style={{
+        width: bounds.width,
+        height: bounds.height,
+      }}
+    >
+      {/* Scrollable container. */}
+      <div className={mx('absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2')}>
+        {children}
+        {items?.map((item, index) => (
```

Comment:
Why does this support both `children` and a higher-order `items`? Usually if the consumer is able to render `children`, it should iterate over `items` on its own so it doesn’t have to reimplement the iterator if the DOM order should change.